drivers/include/sht2x.h

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2016,2017,2018 Kees Bakker, SODAQ
+ * Copyright (C) 2017 George Psimenos
+ * Copyright (C) 2018 Steffen Robertz
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_sht2x SHT2X
+ * @ingroup     drivers_sensors
+ * @brief       Device driver interface for the SHT2X sensor
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the SHT2x temperature and
+ *              humidity sensor.
+ *
+ * @author      Kees Bakker <kees@sodaq.com>
+ * @author      George Psimenos <gp7g14@soton.ac.uk>
+ * @author      Steffen Robertz <steffen.robertz@rwth-aachen.de>
+ */
+
+#ifndef SHT2X_H
+#define SHT2X_H
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief       Status and error return codes
+ */
+enum {
+    SHT2X_OK            =  0,           /**< everything was fine */
+    SHT2X_ERR_I2C       = -1,           /**< error initializing the I2C bus */
+    SHT2X_ERR_NODEV     = -2,           /**< did not detect SHT2x */
+    SHT2X_ERR_NOCAL     = -3,           /**< could not read calibration data */
+    SHT2X_ERR_I2C_READ  = -4,           /**< I2C read error */
+    SHT2X_ERR_TIMEDOUT  = -5,           /**< timed out */
+    SHT2X_ERR_CRC       = -6,           /**< CRC error */
+    SHT2X_ERR_USERREG   = -7,           /**< cannot write User Reg */
+    SHT2X_ERR_RES       = -8,           /**< invalid resolution */
+    SHT2X_ERR_OTHER     = -999,         /**< fatal error */
+};
+
+/**
+ * @brief       Devices supported by this driver
+ */
+typedef enum {
+    SHT2X_TYPE_SHT20    = 0x01,         /**< SHT20 */
+    SHT2X_TYPE_SHT21    = 0x02,         /**< SHT21 */
+    SHT2X_TYPE_SHT25    = 0x03,         /**< SHT25 */
+} sht2x_type_t;
```

Comment:
This isn't used anywhere, should we remove it?

Reply:
It's removed now